### PR TITLE
Rework the way we use Kafka

### PR DESCRIPTION
**Pull request by ctron**
This PR includes a bunch of changes in the way we use Kafka:

* Provision per-application Kafka topics for events automatically – This allows to properly process application data, without the need to filter through all events of all applications. It also introduces a persistent work queue, for which I have mixed feelings.
* Directly consume commands from Kafka – This fixes the problem that only one pod of an endpoint would receive a command, assuming you have scaled up the number of pods
* Directly consume device registry events from Kafka – This allowed to drop a few more dependencies, as most of the code was already present from the command consumer implementation. It also means that we no longer require KNative for running a core Drogue Cloud instance. We also save some more RAM on not running instances for the KafkaSources. Still Knative is being used on the application side.

### CODE REVIEW

File: endpoint-common/src/command/source.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                                log::info!("Failed to dispatch command: {}", err);
+                            }
+                        }
+                        Err(_) => {
+                            log::info!("Failed to convert event to command");
+                        }
+                    },
+                    Err(err) => {
+                        log::info!("Failed to read next event: {}", err);
+                    }
```
Comment (jcrossley3) [inline endpoint-common/src/command/source.rs:60]: Should these bubble up? Or you wanna deal with that later?

**Review by jcrossley3 — COMMENTED** [review]
Man, this is a big change! I'm just gonna trust you, the compiler, and CI for now. :)

I like the general approach, though. Can you elaborate on your "mixed feelings" about the persistent work queue?

File: endpoint-common/src/command/mod.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        Self {
+            device_id,
+            command: command.into(),
+            payload,
+        }
+    }
+}
+
+impl TryFrom<Event> for Command {
+    type Error = ();
```
[the remark this answers is not in the document]
Comment (ctron) [reply endpoint-common/src/command/mod.rs:32]: A bit. But I can fail, with the type `()` as error. I can take a look if I can provide some more reasonable error information.

File: endpoint-common/src/command/source.rs (hunk 1 above)
Comment (ctron) [reply endpoint-common/src/command/source.rs:60]: Well, there is nothing "above" it. As it is running as tokio task, no one else would care.

My idea was (and maybe I should write that in the source code) … if there is a command which cannot be handled, that can happen. And we just discard it. And someone would need to re-try.

File: endpoint-common/src/command/source.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                }
+            }
+            log::info!("Exiting event loop!");
```
Comment (ctron) [inline endpoint-common/src/command/source.rs:63]: But here something is missing … it should set the "alive" flag to "false".

File: endpoint-common/src/command/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ctron) [reply endpoint-common/src/command/mod.rs]: I dug a bit into this code. Should be improved now.

**Comment by ctron** [comment]
> Can you elaborate on your "mixed feelings" about the persistent work queue?

It works. It focuses and what is necessary. There are probably some bugs in there, which we can fix. … Still, it feels a bit like implementing a broker.

**Pull request merged**